### PR TITLE
Feat authorization

### LINKED ISSUE #6: Authentication
`@PreAuthorize` and fetch `User` from jwt principal.

### PR DESCRIPTION
**Pull request by magley**
Closes #6

### CODE REVIEW

File: back/sssscs/src/main/java/com/ib/user/UserController.java [hunk 1]
```diff
@@ -45,9 +50,12 @@ public ResponseEntity<String> login(@Valid @RequestBody UserLoginDto dto) {
 		} catch (BadCredentialsException ex) {
 			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Wrong email or password!");
 		}
+		SecurityContext sc = SecurityContextHolder.getContext();
+		sc.setAuthentication(auth);
```
Comment (elkcert) [inline back/sssscs/src/main/java/com/ib/user/UserController.java]: Unnecessary

**Review by elkcert — CHANGES_REQUESTED** [review]
General notes:
`SecurityConfig`'s `filterChain` does `matcherRegistry.anyRequest().permitAll()`, which should be changed
`SecurityContext`'s `Principal` would be more conveniently accessed through a "facade" class that we inject in every controller where it's required. This would make testing more convenient and decouples the code. https://www.baeldung.com/get-user-in-spring-security#interface

File: back/sssscs/src/main/java/com/ib/user/UserController.java (hunk 1 above)
Comment (magley) [reply back/sssscs/src/main/java/com/ib/user/UserController.java]: Fixed in [d1743de](https://github.com/magley/sssscs/pull/14/commits/d1743de81b68f2e86d68c45a6f8020838cadc3ee).

**Comment by magley** [comment]
> General notes: 
`SecurityContext`'s `Principal` would be more conveniently accessed through a "facade" class that we inject in every controller where it's required. This would make testing more convenient and decouples the code. https://www.baeldung.com/get-user-in-spring-security#interface

Done in [d1743de](https://github.com/magley/sssscs/pull/14/commits/d1743de81b68f2e86d68c45a6f8020838cadc3ee).

**Comment by magley** [comment]
> General notes: `SecurityConfig`'s `filterChain` does `matcherRegistry.anyRequest().permitAll()`, which should be changed

Suggestions?

**Comment by elkcert** [comment]
> > General notes: `SecurityConfig`'s `filterChain` does `matcherRegistry.anyRequest().permitAll()`, which should be changed
> 
> Suggestions?

Shouldn't we require all endpoints to be authenticated, then give permissions on a case-by-case basis?

**Comment by magley** [comment]
> > > General notes: `SecurityConfig`'s `filterChain` does `matcherRegistry.anyRequest().permitAll()`, which should be changed
> > 
> > 
> > Suggestions?
> 
> Shouldn't we require all endpoints to be authenticated, then give permissions on a case-by-case basis?

Added in [358b4cf](https://github.com/magley/sssscs/pull/14/commits/358b4cf5469b486603423b5bbd0620674ca80886).

**Review by elkcert — APPROVED** [review]

**Pull request merged**